extensions/twenty/src/services/TwentySDK.ts

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { getPreferenceValues } from "@raycast/api";
+import { Api } from "../enum/api";
+import fetch from "node-fetch";
+import { getActiveDataModelsSchema } from "./zod/schema/dataModelSchema";
+import { dataModelsFieldsSchema } from "./zod/schema/recordFieldSchema";
+
+class TwentySDK {
+  private url!: string;
+  private token!: string;
+
+  constructor() {
+    const { token, url }: { token: string; url: string } = getPreferenceValues();
+    this.token = `Bearer ${token}`;
+    this.url = url ? `${url}/rest` : `https://api.twenty.com/rest`;
+  }
+
+  async getActiveDataModels() {
+    const response = await fetch(this.url + "/metadata/objects", {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        [Api.KEY]: this.token,
+      },
+    });
+    const rawData = await response.json();
+    const data = getActiveDataModelsSchema.parse(rawData);
```

Comment:
When API Token is invalid this results in "Unexpected end of JSON input" as there was a `500 Internal Server Error` for me.

Reply:
@xmok how should we handle ? should we have a empty view (which let user know this) with action and that action opens up extension preference ?

Reply:
The issue is the error I got is `500` rather than `401` so it's not really evident the issue is invalid token. I suppose one benefit of using an "ErrorView" is we would be able to re-use it for other errors so this works :)
Otherwise, for now, it would also be sufficient to use `primaryAction` of the `Toast` itself. It's up to you.

Reply:
We will have errorview later handling it from toast as of now.

Reply:
I do get the `Toast` but it also gives me an endless loading indicator if API token is wrong. See video with invalid token:

https://github.com/user-attachments/assets/8d3a2b29-cfd8-4086-9dac-f9375a64260c

There are a couple of solutions:
1. remove the new lines you added:
```
if (error) {
    return <></>;
 }
```
this should then just show the default `EmptyView`.

2. OR change the lines to be `Detail` or `List.EmptyView` e.g. 
```
if (error) {
    return <Detail markdown={` # ERROR \n\n ${error.message}`} />;
 }
```